src/api/weather/weatherService.ts

```diff
@@ -0,0 +1,45 @@
+import { env } from '@/common/utils/envConfig';
+
+// Define the shape of the weather data
+interface WeatherData {
+  city: string;
+  temperature: number;
+  weatherPrediction: string;
+}
+
+async function fetchWeather(cityName: string): Promise<WeatherData> {
+  const { SERVICE_WEATHER_URL, SERVICE_WEATHER_SECRET } = env;
+
+  if (!SERVICE_WEATHER_URL || !SERVICE_WEATHER_SECRET) {
+    throw new Error('Weather service configuration is missing.');
+  }
+
+  const apiKey: string = SERVICE_WEATHER_SECRET;
+  const apiURL: string = `${SERVICE_WEATHER_URL}/data/2.5/weather?q=${cityName}&appid=${apiKey}`;
+
+  try {
+    const response = await fetch(apiURL);
+    if (!response.ok) {
+      throw new Error(`Error fetching weather data: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+
+    // Assuming the data structure returned by the API
+    const c: string = data.name; // Adjust according to the actual API response
```

Comment:
I would have done the same thing with setting a schema for the object but ran out of time.